src/gitingest/tests/test_clone.py

```diff
@@ -1,62 +1,132 @@
 from unittest.mock import AsyncMock, patch
 
 import pytest
-
-from gitingest.clone import CloneConfig, check_repo_exists, clone_repo
+from clone import clone_repo, check_repo_exists
+from unittest.mock import patch, AsyncMock
+from asyncio.exceptions import TimeoutError
 
 
 @pytest.mark.asyncio
-async def test_clone_repo_with_commit() -> None:
-    clone_config = CloneConfig(
-        url="https://github.com/user/repo",
-        local_path="/tmp/repo",
-        commit="a" * 40,  # Simulating a valid commit hash
-        branch="main",
-    )
-
-    with patch("gitingest.clone.check_repo_exists", return_value=True) as mock_check:
-        with patch("gitingest.clone.run_git_command", new_callable=AsyncMock) as mock_exec:
-
+async def test_clone_repo_with_commit():
+    query = {
+        "commit": "a" * 40,  # Simulating a valid commit hash
+        "branch": "main",
+        "url": "https://github.com/joydeep049/gitingest",
+        "local_path": "/tmp/repo",
+    }
+
+    with patch("clone.check_repo_exists", return_value=True) as mock_check:
+        with patch(
+            "asyncio.create_subprocess_exec", new_callable=AsyncMock
+        ) as mock_exec:
             mock_process = AsyncMock()
-            mock_process.communicate.return_value = (b"output", b"error")
+            mock_process.communicate.return_value = (b"Cloning Complete", b"")
             mock_exec.return_value = mock_process
-            await clone_repo(clone_config)
-            mock_check.assert_called_once_with(clone_config.url)
+
+            stdout, stdin = await clone_repo(query)
+
+            mock_check.assert_called_once_with(query["url"])
+ 
             assert mock_exec.call_count == 2  # Clone and checkout calls
+            assert b"Cloning Complete" in stdout
 
 
-@pytest.mark.asyncio
-async def test_clone_repo_without_commit() -> None:
-    query = CloneConfig(url="https://github.com/user/repo", local_path="/tmp/repo", commit=None, branch="main")
 
-    with patch("gitingest.clone.check_repo_exists", return_value=True) as mock_check:
-        with patch("gitingest.clone.run_git_command", new_callable=AsyncMock) as mock_exec:
+@pytest.mark.asyncio
+async def test_clone_repo_without_commit():
+    query = {
+        "commit": None,
+        "branch": "main",
+        "url": "https://github.com/joydeep049/gitingest",
```

Comment:
I will let this in the code but be warned that in the future I might remove all references to usernames (including mine) in the codebase

Reply:
Sure!